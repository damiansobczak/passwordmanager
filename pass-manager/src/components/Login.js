import React, { Component } from "react";
import "./Login.scss";
import { Link } from "react-router-dom";

class Login extends Component {
  render() {
    return (
      <div className="card">
        <div className="card-content">
          <div className="card-title">Podaj swoje dane</div>
          <div className="row">
            <div className="col s12">
              <input placeholder="imie@email.com" id="email" type="text" class="validate" />
            </div>
            <div className="col s12">
              <input placeholder="******" id="email" type="text" class="validate" />
            </div>
          </div>
        </div>
        <div className="card-action">
          <Link to={`/passwords`} className="blue-text">
            Zaloguj
          </Link>
        </div>
      </div>
    );
  }
}

export default Login;
