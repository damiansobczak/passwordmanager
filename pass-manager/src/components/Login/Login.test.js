import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Login from "./Login";

describe("<Login />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Login login={() => true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders login information", () => {
    const wrapper = shallow(<Login login={() => true} />);
    expect(wrapper.find(".card-title").text()).toContain("Zaloguj siÄ™ przy pomocy konta Google");
  });

  it("renders login button correctly", () => {
    const wrapper = shallow(<Login login={() => true} />);
    expect(wrapper.find(".btn").text()).toContain("Zaloguj");
  });
});
