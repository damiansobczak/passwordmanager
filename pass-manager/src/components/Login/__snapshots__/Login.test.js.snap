// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col s8 push-s2 mt-5"
    >
      <div
        className="card"
      >
        <div
          className="card-content center-align"
        >
          <div
            className="card-title center-align"
          >
            Zaloguj siÄ™ przy pomocy konta Google
          </div>
          <button
            className="blue accent-4 btn"
            onClick={[Function]}
          >
            Zaloguj
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
